cdap-standalone/src/main/java/io/cdap/cdap/StandaloneMain.java

```diff
@@ -162,6 +162,8 @@ private StandaloneMain(List<Module> modules, CConfiguration cConf) {
     Thread.setDefaultUncaughtExceptionHandler(new UncaughtExceptionHandler());
     this.cConf = cConf;
 
+    cleanupTempDir();
```

Comment:
Can we put a comment here to explain what we expect for this cleanup in constructor.

Reply:
Fixed

Reply:
normally it's not a good idea to perform work in the constructor. I'm also confused how this fixes the problem, is the classloader mentioned in the bug created before Standalone is started?

Reply:
I agree with moving the logic out of the constructor. Currently, StandaloneMain's constructor creates several other classes which eventually instantiates an instance of Authorizer, causing the AuthorizerInstantiator to unpack the provided JAR file into the tmp directory. Since cleanupTempDir is currently called in startup, it is called after the constructor finishes, deleting the loaded classpath.